pkg/assembler/backends/arangodb/path.go

```diff
@@ -0,0 +1,50 @@
+//
+// Copyright 2023 The GUAC Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package arangodb
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/guacsec/guac/pkg/assembler/graphql/model"
+)
+
+func (c *arangoClient) Path(ctx context.Context, source string, target string, maxPathLength int, usingOnly []model.Edge) ([]model.Node, error) {
+	panic(fmt.Errorf("not implemented: Path"))
+}
+
+func (c *arangoClient) Neighbors(ctx context.Context, source string, usingOnly []model.Edge) ([]model.Node, error) {
+	panic(fmt.Errorf("not implemented: Neighbors"))
+}
+
+func (c *arangoClient) Node(ctx context.Context, source string) (model.Node, error) {
+	idSplit := strings.Split(source, "/")
+	if len(idSplit) != 2 {
+		return nil, fmt.Errorf("invalid ID: %s", source)
+	}
+	switch idSplit[0] {
+	case pkgVersionsStr, pkgNamesStr, pkgNamespacesStr, pkgTypesStr:
+		return c.buildPackageResponseFromID(ctx, source, nil)
+	case srcNamesStr, srcNamespacesStr, srcTypesStr:
+		return c.buildSourceResponseFromID(ctx, source, nil)
```

Comment:
So I was going to use this, similar to the `inmem`, where if the ID is provided for packages or sources query, it would default to this (as it is more efficient rather than traversal). In that case, the filter can be used if additional. 

`inmem` implementation: https://github.com/guacsec/guac/blob/main/pkg/assembler/backends/inmem/pkg.go#L442-L458

Reply:
I will make those updates in the follow on PR with the rest of the node query implementations

Reply:
+1. I think likely if we do want to add filtering later on it probably will be implemented on the query level for arango.